core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -235,6 +252,48 @@ public void run() {
             }
         }
 
+        /**
+         * Respond to HTTP request with simple diagnostics.
+         * Primarily used to test the low-level connectivity.
+         */
+        private void respondHello(String header, Socket s) throws IOException {
+            try {
+                Writer o = new OutputStreamWriter(s.getOutputStream(), "UTF-8");
+
+                if (header.startsWith("GET / ")) {
+                    o.write("HTTP/1.0 200 OK\n");
+                    o.write("Content-Type: text/plain;charset=UTF-8\n");
+                    o.write("\n");
+                    o.write("Jenkins-Agent-Protocols: ");
+                    boolean first = true;
+                    for (AgentProtocol p : AgentProtocol.all()) {
+                        if (first)  first = false;
+                        else        o.write(",");
+                        o.write(p.getName());
+                    }
+                    o.write("\n");
+                    o.write("Jenkins-Version: " + Jenkins.VERSION + "\n");
+                    o.write("Jenkins-Session: " + Jenkins.SESSION_HASH + "\n");
+                    o.write("Client: " + s.getInetAddress().getHostAddress() + "\n");
```

Comment:
You are printing "Client" and giving the servers address.

perhaps you want to print "Server"?